src/gt4py/next/ffront/foast_to_itir.py

```diff
@@ -436,144 +225,107 @@ def visit_Call(self, node: foast.Call, **kwargs) -> itir.FunCall | itir.Literal:
                 ts_ffront.ScanOperatorType,
             ),
         ):
-            # operators are lowered into stencils and only accept iterator
-            #  arguments. As such transform all value arguments, e.g. scalars
-            #  and tuples thereof, into iterators. See ADR-0002 for more
-            #  details.
+            # Operators are lowered into lifted stencils.
             lowered_func = self.visit(node.func, **kwargs)
-            lowered_args = []
-            for arg in node.args:
-                lowered_arg = self.visit(arg, **kwargs)
-                if iterator_type_kind(arg.type) == ITIRTypeKind.VALUE:
-                    lowered_arg = im.lift_(im.lambda__()(lowered_arg))()
-                lowered_args.append(lowered_arg)
-
-            return self._lift_if_field(node)(im.call_(lowered_func)(*lowered_args))
+            lowered_args = [self.visit(arg, **kwargs) for arg in node.args]
+            args = [f"__arg{i}" for i in range(len(lowered_args))]
+            return im.lift_(im.lambda__(*args)(im.call_(lowered_func)(*args)))(*lowered_args)
+        elif isinstance(node.func.type, ts.FunctionType):
+            return im.call_(self.visit(node.func, **kwargs))(*self.visit(node.args, **kwargs))
 
         raise AssertionError(
             f"Call to object of type {type(node.func.type).__name__} not understood."
         )
 
     def _visit_astype(self, node: foast.Call, **kwargs) -> itir.FunCall:
-        obj = (to_value(node.args[0]))(self.visit(node.args[0], **kwargs))
-        dtype = self.visit(node.args[1]).id
-        return self._lift_lambda(node)(im.call_("cast_")(obj, dtype))
+        assert len(node.args) == 2 and isinstance(node.args[1], foast.Name)
+        obj, dtype = node.args[0], node.args[1].id
+
+        # TODO check that we test astype that results in a itir.map_ operation
```

Comment:
TODO

Reply:
I added this as a task to the cleanup/refactoring project for next cycle. I could add a test now that specifically tests as_type in a mapped context, but I prefer to setup a test-case that allows testing other kinds of builtins in mapped context as well.